.github/workflows/go.yml

```diff
@@ -8,14 +8,11 @@ jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2.4.0
-        with:
-          fetch-depth: 0
+      - name: Checkout
+        uses: actions/checkout@v2.4.0
 
-      - name: Setup Go toolchain
-        uses: ./actions/setup-go
-        with:
-          cacheKey: build
```

Comment:
Build and release aren't using the same go modules and thus have different requirements for the cache. I would recommend keeping these.

Imagine if `release` ends up creating the cache, then `build` will use the cache created by `release` and will probably have to download either sage tools and/or go modules every time. The way github cache works is that if there is a cache hit, it won't update the cache even if things have changed.

Reply:
In this repo this is most likly not an issue or if it is, a small one.

In other repos with multiple jobs I definitively recommend having unique cache keys per job.

Reply:
Ah, darn - good point